sematic/ui/src/components/ActionMenu.tsx

```diff
@@ -0,0 +1,95 @@
+import { KeyboardArrowDown } from "@mui/icons-material";
+import { Box, Button, Menu, Typography, useTheme } from "@mui/material";
+import React, { MouseEvent, useState } from "react";
+
+export function ActionMenuItem(props: {
+  title: string;
+  enabled?: boolean;
+  onClick?: () => void;
+  titleColor?: string;
+  children?: any;
+  handleClose?: () => void;
+}) {
+  const theme = useTheme();
+  const {
+    title,
+    enabled = true,
+    titleColor = theme.palette.primary.main,
+    children,
+    onClick,
+    handleClose,
+  } = props;
+
+  return (
+    <Button
+      sx={{
+        display: "block",
+        textAlign: "left",
+        px: 5,
+        py: 2,
+        width: "100%",
+        opacity: enabled ? 1 : 0.5,
+      }}
+      disabled={!enabled}
+      onClick={() => {
+        handleClose && handleClose();
+        onClick && onClick();
+      }}
+    >
+      <Typography
+        sx={{
+          color: titleColor,
+          display: "block",
+          fontWeight: "inherit",
+        }}
+      >
+        {title}
+      </Typography>
+      <Box sx={{ color: "GrayText" }}>{children}</Box>
+    </Button>
+  );
+}
+
+export function ActionMenu(props: { title: string; children?: any }) {
+  const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+  const open = Boolean(anchorElement);
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    setAnchorElement(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorElement(null);
+  };
+
+  const { children, title } = props;
+
+  const makeClosable = (element: React.ReactElement<any>) =>
+    React.isValidElement(element)
+      ? React.cloneElement(element as React.ReactElement<any>, {
+          handleClose: handleClose,
+          key: Math.random().toString(),
+        })
+      : element;
```

Comment:
Could you please explain to me what this does like I was 5?

Reply:
Yes.
In order for the menu to close when an item is clicked, the menu items need to have a `handleClose` prop set that hides the menu. This is a feature of the `Menu` MUI component.
Now here `ActionMenu` uses the children components as menu items:
```typescript
<ActionMenu>
    <ActionMenuItem>...</ActionMenuItem>
    <ActionMenuItem>...</ActionMenuItem>
    ...
</ActionMenu>
```
So this piece of code automatically attaches the `handleClose` prop to each child. This is the canonical way to do this according to Lord Internet.